From: Usama Arif <usamaarif642@gmail.com>
Subject: selftests/mm: include linux/mman.h for prctl_thp_disable
Date: Wed, 10 Sep 2025 21:46:09 +0100

MADV_COLLAPSE is part of linux/mman.h and needs to be included for this
selftest for glibc compatibility.  It is also included in other tests that
use MADV_COLLAPSE.

Link: https://lkml.kernel.org/r/20250910204609.1720498-1-usamaarif642@gmail.com
Signed-off-by: Usama Arif <usamaarif642@gmail.com>
Reported-by: Mark Brown <broonie@kernel.org>
Closes: https://lore.kernel.org/all/c8249725-e91d-4c51-b9bb-40305e61e20d@sirena.org.uk/
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 tools/testing/selftests/mm/prctl_thp_disable.c |    1 +
 1 file changed, 1 insertion(+)

--- a/tools/testing/selftests/mm/prctl_thp_disable.c~selftests-prctl-introduce-tests-for-disabling-thps-completely-fix-2
+++ a/tools/testing/selftests/mm/prctl_thp_disable.c
@@ -9,6 +9,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <sys/mman.h>
+#include <linux/mman.h>
 #include <sys/prctl.h>
 #include <sys/wait.h>
 
_
